service/cloudconfig/service.go

```diff
@@ -0,0 +1,98 @@
+package cloudconfig
+
+import (
+	"github.com/giantswarm/azuretpr"
+	k8scloudconfig "github.com/giantswarm/k8scloudconfig"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/viper"
+
+	"github.com/giantswarm/azure-operator/flag"
+)
+
+// Config represents the configuration used to create a cloudconfig service.
+type Config struct {
+	// Dependencies.
+
+	Logger micrologger.Logger
+
+	// Settings.
+
+	Flag  *flag.Flag
+	Viper *viper.Viper
+}
+
+// DefaultConfig provides a default configuration to create a new cloudconfig service
+// by best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		Logger: nil,
+
+		// Settings.
+		Flag:  nil,
+		Viper: nil,
+	}
+}
+
+// Service implements the cloudconfig service interface.
+type Service struct {
+	Config
```

Comment:
I dislike embedding `Config` into the `Service`. It desn't feel right that code outside can mutate the Service.

I'm also on a mission of encouraging people to not call structs `Service`, this makes it difficult to track stacked calls. I'd suggest `CloudConfig` name for this struct.

Reply:
+1

Reply:
Done